plugins/tileacq.py

```diff
@@ -384,17 +381,19 @@ def _move_to_tile(self, idx, orig_pos, tile_size, prev_idx):
         # don't move on the axis that is not supposed to have changed
         main_data = self.main_app.main_data
         prev_pos = main_data.stage.position.value
-        if numpy.argmax(abs(numpy.subtract(idx, prev_idx))) == 0:
-            tile_pos = (orig_pos["x"] + idx[0] * tile_size[0] * overlap,
-                        prev_pos["y"])
-        else:
-            tile_pos = (prev_pos["x"],
-                        orig_pos["y"] - idx[1] * tile_size[1] * overlap)
-
-        logging.debug("Moving to tile %s at %s m", idx, tile_pos)
-        f = self.main_app.main_data.stage.moveAbs({"x": tile_pos[0], "y": tile_pos[1]})
+        m = {}
+        idx_change = numpy.subtract(idx, prev_idx)
+        if idx_change[0]:
+            m["x"] = orig_pos["x"] + idx[0] * tile_size[0] * overlap
+        if idx_change[1]:
+            m["y"] = orig_pos["y"] - idx[1] * tile_size[1] * overlap
+
+        logging.debug("Moving to tile %s at %s m", idx, m)
+        f = self.main_app.main_data.stage.moveAbs(m)
         try:
-            f.result(100)
+            speed = 10e-6  # m/s. Assume very low speed for timeout.
+            t = max(tile_size[0] * overlap, tile_size[1] * overlap) / speed
```

Comment:
for the distance of the move, don't use max(), by math.hypoth(), that's a little bit more mathematically correct, and more pessimistic. Also, add 1 s , to be sure to never timeout in case of very tiny move.